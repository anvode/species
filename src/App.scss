.main {
    background-color: white;
    flex: 1 0 auto;
    margin-left: auto;
    margin-right: auto;
    max-width: 1200px;
    padding: 0;
    width: 100%;

    &__title {
        background-color: var(--color-primary);
        margin: 0;
        padding: 1rem 0;
        text-align: center;
    }
}
