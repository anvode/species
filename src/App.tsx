import * as React from 'react';

import SpeciesProvider from './context/species/SpeciesContext';
import Species from './components/Species';
import PageTitle from './components/PageTitle/Pagetitle';

import './App.scss';

interface AppProps {}

function App(props: AppProps) {
    return (
        <div className="main">
            <SpeciesProvider>
                <PageTitle></PageTitle>
                <div className="container">
                    <Species></Species>
                </div>
            </SpeciesProvider>
        </div>
    );
}

export default App;

