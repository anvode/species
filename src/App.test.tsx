import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from './App';

afterEach(cleanup);

describe('App', () => {
    test('renders title', () => {
        const { getByText } = render(<App />);
        const title = getByText(/Species/i);
        expect(title).toBeInTheDocument();
    });

    it('snapshot App', () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});

