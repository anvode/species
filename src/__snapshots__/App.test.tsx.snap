// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot App 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <h1
      class="main__title"
    >
      IUCN Red List of Threatened Species
      <strong />
    </h1>
    <div
      class="container"
    >
      <div
        class="loader__container"
      >
        <h2
          class="loader__title"
        >
          Fetching: 
          <strong>
            Regions
          </strong>
        </h2>
        <div
          class="loader"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
